### PR TITLE
feat: ahbap map

### LINKED ISSUE #739: feat: Import ahbap google maps data and show it as another layer on map
## Define your request
We want to import the data ahbap uses in google maps from: https://www.google.com/maps/d/u/0/viewer?mid=1aQ0TJi4q_46XAZiSLggkbTjPzLGkTzQ&hl=tr&fbclid=PAAabyVDiTmFsFSPkcoAfAd9pcy5poekIm5LfUOndiuxmCXEMq4By3ObMr-_8&ll=0%2C0&z=6

**discord username: `umut#3072`**

Strategy
- [ ] Create an api route at next.js backend and fetch `kml`/`kmz` data for the above map.
  - [ ] Use an in-memory cache that invalidates itself every 5 minutes.
  - [ ] Serve the kml/kmz files from this in-memory cache so we will hit google maps every 5 mins (as opposed to hitting it with every request)

### PR DESCRIPTION
**Pull request by usirin**
# Description

WIP implementation of ahbap data that fetches the kmz data and turns it into a geojson object, before passing it down to the client

note that this is still pretty much very WIP and i think this data needs to come through our data/backend pipeline under a channel named "ahbap"

cc @acikkaynak/afet-harita-backend 

**discord username: `umut#3072`**

**closes #739**

Please describe your changes. Also describe your aim and content. Do not forget to list the dependencies required caused by those changes.

## Things to check before creating a PR
- [ ] I have inspected my topic, checked.
- [ ] If it is a core feature, executed detailed tests.

## Creating PR rules
- [ ] PR must be created for an issue with approved tag. Otherwise PR will be rejected.
- [ ] Relevant issue number: The issue number related to PR must be attached to head of PR header, after prefix after prefix # must be attached in parenthesis. A header like this could be used "prefix(#issue_number): PR header" .
- [ ] A descriptive and understandable title: The PR title should clearly describe the nature and purpose of the changes. The PR title should be the first thing displayed when the PR is opened. And it should follow the semantic commit rules. For example, a title like "docs(#issueId): Add README.md" can be used.
- [ ] Related file selection: Only relevant files should be touched and no other files should be affected.
- [ ] Format and Lint Suitability : The code should be made in accordance with a certain format standard and examined according to the lint rules.
- [ ] Clean commit history: The commits where the changes are made should be clear and organized.
- [ ] Opening PR when job is completed: PR should be opened when job is completed and sent for review by other team members.

## Changes

- [ ] A new feature (a change that adds a new feature, not a breaking change)
- [ ] A new refactor (a change that is not a breaking change, that improves the readability or performance of the code)
- [ ] A breaking change
- [ ] Documentation change

# How were these changes tested?

Please describe the tests you did to test the changes you made. Please also specify your test configuration.

**Test Configuration**:

* Firmware version:
* Hardware:
* Toolchain:
* SDK:

### CODE REVIEW

**Comment by Ertan222** [comment]
@usirin Can you solve conflicts ?

**Comment by usirin** [comment]
@Ertan222 this is a wip PR, we can take it easy on this.

File: afet-harita.d.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+declare module "parse2-kmz" {
```
Comment (fdemir) [inline afet-harita.d.ts:1]: why do we need this?

File: components/UI/Drawer/components/channels/twitter/FeedChannelAhbap.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import Typography from "@mui/material/Typography";
+import { FeedChannelAhbapProps } from "../../types";
+import { Divider } from "@mui/material";
+
+const getTypeLabel = (type: string) => {
+  const id = type.split("-")[1];
+
+  switch (id) {
+    case "1507":
```
Comment (fdemir) [inline components/UI/Drawer/components/channels/twitter/FeedChannelAhbap.tsx:9]: can be stored key: value object?

```
const findANameForThis = {
 1826: "Guvenli Noktalar"
}
```

Comment (fdemir) [reply components/UI/Drawer/components/channels/twitter/FeedChannelAhbap.tsx:9]: or enum maybe

File: hooks/useUrlPath.ts [hunk 3]
```diff
@@ -8,8 +8,12 @@ export function useUrlPath() {
     const query = new URLSearchParams();
     const modifiedArray = Object.entries(modifiedQuery);
     modifiedArray.forEach((entry) => {
-      query.append(entry[0], entry[1].toString());
+      if (entry[0] && entry[1]) {
+        query.append(entry[0], entry[1].toString());
+      }
     });
+    console.log(query.toString(), ">>>");
```
Comment (fdemir) [inline hooks/useUrlPath.ts:15]: remove this one if we do not need

**Comment by usirin** [comment]
thanks for the review @fdemir, though it's still wip, opened it to show progress to people. will remove `wip` label when it's ready for review.

**Comment by eraygundogmus** [comment]
> thanks for the review @fdemir, though it's still wip, opened it to show progress to people. will remove `wip` label when it's ready for review.

can you please make this PR draft while you work?

**Comment by dopplerDistortion** [comment]
@usirin Seperation of marker types feels weird when they are clustered but acceptable. If we don't have and restrictions, we can:
- Visually differentiate clusters for Ahbap Data and Requests
or
- Group all marker types in single cluster type.

https://user-images.githubusercontent.com/1084392/218262494-e2fb8749-0f95-428d-95ab-3c0d500b8364.mov

**Comment by fdemir** [comment]
ill handle this issue, collabrating with be. gonna make some changes. thx.

**Comment by fdemir** [comment]
small update for this pr:

currently this pr fetching the ahbap places by parsing the google XML file. the be team is going to work about storing data at db. we will remove the xml parser and fetch over new ep.

File: package.json [hunk 4]
```diff
@@ -40,6 +40,7 @@
     "formatcoords": "^1.1.3",
     "i18next": "^22.4.9",
     "leaflet": "^1.9.3",
+    "leaflet-kmz": "^1.0.9",
```
Comment (cdagli) [inline package.json:43]: Seems to be not used anymore?

File: pages/index.tsx [hunk 5]
```diff
@@ -18,7 +18,11 @@ import {
   useDevice,
 } from "@/stores/mapStore";
 import styles from "@/styles/Home.module.css";
-import { REQUEST_THROTTLING_INITIAL_SEC } from "@/utils/constants";
+import {
+  AHBAP_LOCATIONS_URL,
```
Comment (cdagli) [inline pages/index.tsx:22]: We also have a file named `urls.ts`

File: pages/index.tsx [hunk 6]
```diff
@@ -122,6 +128,37 @@ export default function Home({ deviceType, singleItemDetail }: Props) {
     }
   );
 
+  useSWR(AHBAP_LOCATIONS_URL, dataFetcher, {
+    onSuccess: (data) => {
+      if (!data) return;
+      console.log(data.results);
```
Comment (cdagli) [inline pages/index.tsx:134]: Remove?

File: pages/index.tsx [hunk 7]
```diff
@@ -122,6 +128,37 @@ export default function Home({ deviceType, singleItemDetail }: Props) {
...
+          type: "Feature",
+          geometry: {
+            type: "Point",
+            coordinates: item.loc,
+          },
+          properties: extra_params,
+        };
+      });
+
+      console.log(features);
```
Comment (cdagli) [inline pages/index.tsx:156]: Remove?

File: components/UI/Map/AhbapClusterGroup.tsx [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+        if (isCluster) {
+          return (
+            <Marker
+              key={`cluster-${cluster.id}`}
+              position={[latitude, longitude]}
+              icon={fetchIcon(pointCount)}
+              eventHandlers={{
+                click: (e) => {
+                  console.log(e, cluster);
```
Comment (cdagli) [inline components/UI/Map/AhbapClusterGroup.tsx:57]: Let's remove all these `console.logs` in a follow up PR.

File: components/UI/Map/AhbapClusterGroup.tsx [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+            key={cluster.properties.id}
+            position={[latitude, longitude]}
+            icon={L.icon({
+              iconUrl: cluster.properties.icon || emptyIcon,
+              iconSize: [28, 28],
+              iconAnchor: [14, 14],
+            })}
+            eventHandlers={{
+              click: (e) => {
+                console.log(e, cluster);
```
Comment (OrhanOzkercin) [inline components/UI/Map/AhbapClusterGroup.tsx:82]: log kalmis

**Pull request merged**